Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM php:7.2-fpm-alpine
+
+# Use the default production configuration
+RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
+
+RUN apk add --no-cache curl git tar unzip libpng-dev libxml2-dev
+
+RUN docker-php-ext-install bcmath && \
+    docker-php-ext-install ctype && \
+    docker-php-ext-install json && \
+    docker-php-ext-install gd && \
+    docker-php-ext-install mbstring && \
+    docker-php-ext-install pdo && \
+    docker-php-ext-install pdo_mysql && \
+    docker-php-ext-install tokenizer && \
+    docker-php-ext-install xml && \
+    curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer && \
```

Comment:
Instead of using `curl`, I would actually grab composer from the composer docker image by using a multi-stage build.

Reply:
Troubleshooting > PHP version & extensions in https://hub.docker.com/_/composer/ says:
> Our image is aimed at quickly running Composer without the need for having a PHP runtime installed on your host. You should not rely on the PHP version in our container. We do not provide a Composer image for each supported PHP version because we do not want to encourage using Composer as a base image or a production image.

Reply:
The issue with `curl` is, you're not checking that the file its downloading is the one that Composer provides (no checksum verification).

By using a multi-stage build, you can just grab the `composer` executable from the Composer docker image and use that, which requires no external `curl` calls and can be cached by Docker.

Reply:
Just read about multi-stage build. TLDR it enables to grab artifacts from an existing build. Looks pretty good. Gotta figure out how to do that.

Reply:
Should just be able to add this line into your `Dockerfile`:

```
COPY --from=composer:latest /usr/bin/composer /usr/local/bin/composer
```